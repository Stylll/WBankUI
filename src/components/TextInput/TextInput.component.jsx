import React from 'react';
import PropTypes from 'prop-types';

import './TextInput.styles.scss';

const TextInput = ({
  id, name, value, title, type, placeholder,
  onChange, info, error, required, dataTestId,
}) => {
  const newPlaceholder = (required) ? `${placeholder} *` : placeholder;
  return (
    <div className="form-input">
                <div className="title">{title}</div>
                <input type={type} name={name} data-testid={dataTestId}
                    id={id} className="text-input" placeholder={newPlaceholder}
                    onChange={onChange} value={value}
                />
          {!error && <span className="info">{info}</span>}
          {error && <span className="error" id={`${name}-error`}>{error}</span>}
</div>
  );
};

// proptypes
TextInput.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  title: PropTypes.string,
  type: PropTypes.string.isRequired,
  info: PropTypes.string,
  error: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  required: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  dataTestId: PropTypes.string,
};

TextInput.defaultProps = {
  id: '',
  title: '',
  info: '',
  required: false,
  dataTestId: 'textinput',
};

export default TextInput;
