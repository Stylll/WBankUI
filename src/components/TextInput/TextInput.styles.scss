@import '../../partials/colors';

.form-input {
    width: 100%;
    margin: 30px 0px;
    .title {
      text-transform: capitalize;
      color: $primary-color;
      font-size: 14px;
      text-align: left;
    }
    .text-input {
      width: 100%;
      height: 30px;
      border: 0px;
      border-bottom: 1px solid $gray;
      border-radius: 0;
      outline: none;
      font-size: 14px;
      color: $black;
      padding: 2px 4px;
      &:focus {
        border-color: $primary-color;
      }
    }
    .info {
      font-size: 12px;
      color: $gray;
    }
    .error {
      font-size: 12px;
      color: $red;
    }
  }
