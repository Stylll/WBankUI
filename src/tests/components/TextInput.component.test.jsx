import { render, fireEvent } from '@testing-library/react';
import React from 'react';

import TextInput from '../../components/TextInput/TextInput.component';

describe('TextInput Component', () => {
  it('should render properly', () => {
    const { baseElement } = render(<TextInput
        name="fullname"
        title="Fullname"
        value="Johnson"
        type="text"
        error=""
        placeholder="enter your name"
        onChange={() => {}}
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should render properly with error message', () => {
    const { baseElement } = render(<TextInput
        name="fullname"
        title="Fullname"
        value="Johnson"
        type="text"
        error="fullname is required"
        placeholder="enter your name"
        onChange={() => {}}
        />);

    expect(baseElement).toMatchSnapshot();
  });

  it('should call onchange method when value is typed', () => {
    const onChange = jest.fn();
    const { getByTestId } = render(<TextInput
        name="fullname"
        title="Fullname"
        value="Johnson"
        type="text"
        error="fullname is required"
        placeholder="enter your name"
        onChange={onChange}
        dataTestId="textinput"
        />);

    fireEvent.change(getByTestId('textinput'), { target: { value: '123' } });
    expect(onChange).toHaveBeenCalled();
  });
});
