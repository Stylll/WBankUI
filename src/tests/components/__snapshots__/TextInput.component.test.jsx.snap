// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Component should render properly 1`] = `
<body>
  <div>
    <div
      class="form-input"
    >
      <div
        class="title"
      >
        Fullname
      </div>
      <input
        class="text-input"
        data-testid="textinput"
        id=""
        name="fullname"
        placeholder="enter your name"
        type="text"
        value="Johnson"
      />
      <span
        class="info"
      />
      
    </div>
  </div>
</body>
`;

exports[`TextInput Component should render properly with error message 1`] = `
<body>
  <div>
    <div
      class="form-input"
    >
      <div
        class="title"
      >
        Fullname
      </div>
      <input
        class="text-input"
        data-testid="textinput"
        id=""
        name="fullname"
        placeholder="enter your name"
        type="text"
        value="Johnson"
      />
      <span
        class="error"
        id="fullname-error"
      >
        fullname is required
      </span>
    </div>
  </div>
</body>
`;
